src/libs/Trie/TrieNode.js

```diff
@@ -0,0 +1,50 @@
+
+class TrieNode {
+    constructor() {
+        this.children = {};
+        this.leaf = false;
+        this.metaData = {};
+    }
+
+    /**
+     * Make the current as a leaf node.
+    */
+    setCompleteWord() {
+        this.leaf = true;
+    }
+
+    /**
+    * Check if the current node is a leaf node.
+    * @returns {Boolean}
+    */
+    isCompleteWord() {
```

Comment:
We should not mix the terminology of `isCompleteWord` and `isLeaf`. We should choose one or the other and be consistent throughout the code

Reply:
all getters and setters methods have been removed.